spec/dsl_spec.rb

```diff
@@ -0,0 +1,33 @@
+require "spec_helper"
+
+RSpec.describe "Pundit RSpec DSL" do
+  let(:fake_rspec) do
+    double = class_double(RSpec::ExampleGroups)
+    double.extend(::Pundit::RSpec::DSL)
+    allow(double).to receive(:describe).and_yield
+    double
+  end
+  let(:block) { proc { "block content" } }
+
+  it "calls describe with the correct metadata and without :focus" do
+    expect { |block| fake_rspec.permissions(:item1, :item2, &block) }.to yield_control
```

Comment:
Thanks for making the changes, it's looking good!

I have but one small nitpick left, which I myself caused by not being clear enough in my previous comment. Sorry!

I believe you should remove line 7 (`allow(double).to receive(:describe)`), line 13 (`expect { ... }.to yield_control`), and line 24 (`expect { ... }.to yield_control` }). They don't add anything to the test, and they make for more reading material.

It's not strictly important why I suggested these, but I'll explain my reasoning for the curious:
* Line 7 allows us to change `expect(fake_rspec).to receive(:describe)` into `expect(fake_rspec).to have_received(:describe)`, which means we can collect all assertions at the end of the test. I find the test writing style of "1. setup, 2. execute, 3. assert" to work really well in general, so I gravitate towards that.
* Line 13 and 24 were intended as an alternative to the bare `fake_rspec.permissions(...)`, but honestly I think asserting the block is called in our expectation around `describe` is more useful.

Reply:
@Burgestrand sorry for misunderstanding your suggestions and thank you for clarifying! I really enjoyed the explanations. hopefully these latest changes are satisfactory! 😅